Modules/_curses_panel.c

```diff
@@ -429,21 +429,30 @@ PyCursesPanel_New(_curses_panel_state *state, PANEL *pan,
     return (PyObject *)po;
 }
 
+static int
+PyCursesPanel_Clear(PyObject *op)
+{
+    PyCursesPanelObject *self = _PyCursesPanelObject_CAST(op);
+    PyObject *extra = (PyObject *)panel_userptr(self->pan);
+    if (extra != NULL) {
+        Py_CLEAR(extra);
+        if (set_panel_userptr(self->pan, NULL) == ERR) {
+            curses_panel_panel_set_error(self, "set_panel_userptr", NULL);
+            PyErr_FormatUnraisable("Exception ignored in tp_clear of %T", op);
+        }
+    }
+    // do NOT clear self->wo yet as there is no cycle to break with it
```

Comment:
The reason is that we need an associated WINDOW to be live until the panel is deleted.